### PR TITLE
Introduce test tiers

### PR DESCRIPTION
**Pull request by aufi**
To provide maximum information about the project functionality, tests exections were separated into three tiers (stages) 0,1,2. From core functionality to nice to haves.

Updated github workflows, Makefile targets and README with description.

### CODE REVIEW

**Comment by aufi** [comment]
Stage1 target depends on https://github.com/konveyor/go-konveyor-tests/pull/16

**Comment by aufi** [comment]
@mguetta1 @sshveta @dymurray @fabianvf Hi, please take a look, I think it makes sense split test executions in this repo to stages to have (rock) solid stage 0, solid stage 1 and maybe experimental stage 2. More details are in README and Makefile. I'm interested in your feedback, thanks!

**Comment by fabianvf** [comment]
@aufi This looks good to me, were there any tests that needed to be dropped or are we at test parity with the old analyzer now?

**Review by mguetta1 — CHANGES_REQUESTED** [review]
Thanks @aufi! I added few comments

**Comment by aufi** [comment]
> @aufi This looks good to me, were there any tests that needed to be dropped or are we at test parity with the old analyzer now?

We had a very little of tests from past on analysis (Pathfinder example-1 app and Petclinic). Currently Petclinic w/hazelcast is not included in analysis test - will be added (in one of next PRs).

File: Makefile [hunk 1]
```diff
@@ -17,13 +17,40 @@ setup:
...
+
+# Application analysis tests.
 test-analysis:
 	HUB_BASE_URL="http://${MINIKUBE_IP}/hub" go test -count=1 -timeout 7200s -v ./analysis/...
 
-# Execute all tests.
-test-all: test-analysis
+# Metrics.
+test-metrics:
+	HUB_BASE_URL="http://${MINIKUBE_IP}/hub" go test -count=1 -v ./metrics/...
```
[the remark this answers is not in the document]
Comment (mguetta1) [reply Makefile:51]: I see, thanks

**Review by mguetta1 — APPROVED** [review]
LGTM

**Comment by aufi** [comment]
Merging with passing tier0.

**Pull request merged**